Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -504,3 +510,206 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
+ DEALINGS IN THE SOFTWARE
+ */
+
+/**
+ A type-erased `Encodable` value.
+ The `AnyEncodable` type forwards encoding responsibilities
+ to an underlying value, hiding its specific underlying type.
+ You can encode mixed-type values in dictionaries
+ and other collections that require `Encodable` conformance
+ by declaring their contained type to be `AnyEncodable`:
+     let dictionary: [String: AnyEncodable] = [
+         "boolean": true,
+         "integer": 42,
+         "double": 3.141592653589793,
+         "string": "string",
+         "array": [1, 2, 3],
+         "nested": [
+             "a": "alpha",
+             "b": "bravo",
+             "c": "charlie"
+         ],
+         "null": nil
+     ]
+     let encoder = JSONEncoder()
+     let json = try! encoder.encode(dictionary)
+ */
+private struct MetricEncodable: Encodable {
+    let value: Any
+
+    init<T>(_ value: T?) {
+        self.value = value ?? ()
+    }
+
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+
+        switch value {
+        #if canImport(Foundation)
+        case let number as NSNumber:
+            try encode(nsnumber: number, into: &container)
+        case is NSNull:
+            try container.encodeNil()
+        #endif
+        case is Void:
+            try container.encodeNil()
+        case let bool as Bool:
+            try container.encode(bool)
+        case let int as Int:
+            try container.encode(int)
+        case let int8 as Int8:
+            try container.encode(int8)
+        case let int16 as Int16:
+            try container.encode(int16)
+        case let int32 as Int32:
+            try container.encode(int32)
+        case let int64 as Int64:
+            try container.encode(int64)
+        case let uint as UInt:
+            try container.encode(uint)
+        case let uint8 as UInt8:
+            try container.encode(uint8)
+        case let uint16 as UInt16:
+            try container.encode(uint16)
+        case let uint32 as UInt32:
+            try container.encode(uint32)
+        case let uint64 as UInt64:
+            try container.encode(uint64)
+        case let float as Float:
+            try container.encode(float)
+        case let double as Double:
+            try container.encode(double)
+        case let string as String:
+            try container.encode(string)
+        #if canImport(Foundation)
+        case let date as Date:
+            try container.encode(date)
+        case let url as URL:
+            try container.encode(url)
+        #endif
+        case let array as [Any?]:
+            // DD Logs app fails to render arrays of JSON.
+            // Here we map the array to a dictionary with indexes as keys.
+            var dictionary: [String: MetricEncodable] = [:]
+            array.enumerated().forEach { dictionary["\($0.offset)"] = MetricEncodable($0.element) }
+            try container.encode(dictionary)
```

Comment:
Do we need the entire "Flight-School/AnyCodable" implementation instead of our `CodableValue` only because of this trick? If so, wouldn't it be simpler to just trick our [`CodableValue`](https://github.com/DataDog/dd-sdk-ios/blob/ad16182bba3f757406834044bf0d9d1fdcdd3dea/Sources/Datadog/Core/Utils/CodableValue.swift) with external modifier?

```swift
private func flattenArraysToObjects(in codableValue: inout CodableValue) -> CodableValue {
   if let codableArray = codableValue.value as? [CodableValue] {
      codableValue.value = [
         "0": codableArray[0],
         "1": codableArray[1],
         // ... (just smarter)
      ]
   }

   // recursive flattenArraysToObjects(...)

   return codableValue
}
```

This might require making [`codableValue.value`](https://github.com/DataDog/dd-sdk-ios/blob/ad16182bba3f757406834044bf0d9d1fdcdd3dea/Sources/Datadog/Core/Utils/CodableValue.swift#L20) mutable `var`, which is not a big deal for `dogfooding` branch.

Reply:
Here I'm using the dictionary representation of the payloads with `.dictionaryRepresentation()`, which are of type `[String: Any]`. Using `AnyEncodable` is more straightforward and avoid having to decode the data returned from `.jsonRepresentation()` and then map it to transform the arrays. Note that this is only a small piece of the `AnyEncodable` from `Flight-School/AnyCodable` and `AnyEncodable` support all primitives supported by `Encodable`.

Also, the modification is self-contained in a single file to avoid any potential conflicts on the branch. I'm afraid that having to modify `CodableValue` would increase the impact.

Reply:
I see 👌, indeed this way we don't need to decode JSON string and we're supporting more `Foundation` types 👍.